plugins/linkattacher/src/linkattacherserverimpl.cpp

```diff
@@ -16,147 +16,155 @@ LinkAttacherServerImpl::LinkAttacherServerImpl()
+    gazebo::physics::ModelPtr child_model = _world->ModelByName(child_model_name);
     #else
-    gazebo::physics::ModelPtr robot_model = _world->GetModel(robot_name);
+    gazebo::physics::ModelPtr child_model = _world->GetModel(child_model_name);
     #endif
 
-    if(!robot_model)
+    if(!child_model)
     {
-        yError() << LogPrefix << "Attach: " << robot_name << " does not exist in gazebo";
+        yError() << LogPrefix << "Attach: " << child_model_name << " model does not exist in gazebo";
         return false;
     }
     else
     {
-      yInfo() << LogPrefix << "Attach: " << robot_model->GetName() << " found";
+      yInfo() << LogPrefix << "Attach: " << child_model->GetName() << " model found";
     }
 
     //Get the exact link with only link name instead of full_scoped_link_name
-    gazebo::physics::Link_V robot_model_links = robot_model->GetLinks();
-    for(int i=0; i < robot_model_links.size(); i++)
+    gazebo::physics::Link_V child_model_links = child_model->GetLinks();
+    gazebo::physics::LinkPtr child_model_link;
+    for(int i=0; i < child_model_links.size(); i++)
     {
-        //E.g iCub::l_hand::l_hand_base_link
-
-        std::string candidate_robot_link_name = robot_model_links[i]->GetScopedName();
+        std::string candidate_child_model_link_name = child_model_links[i]->GetScopedName();
 
-        std::size_t lastcolon = candidate_robot_link_name.rfind(":");
-        std::string unscoped_robot_link_name = candidate_robot_link_name.substr(lastcolon+1,std::string::npos);
-
-        if(unscoped_robot_link_name == robot_link_name)
+        if(GazeboYarpPlugins::hasEnding(candidate_child_model_link_name, child_model_link_name))
         {
-            yInfo() << LogPrefix << "Attach: Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
-            gazebo::physics::LinkPtr robot_link = robot_model_links[i];
-            if(!robot_link)
-            {
-                yError() << LogPrefix << "Attach: " << robot_link_name << " is not found";
-                return false;
-            }
-            else
-            {
-              yInfo() << LogPrefix << "Attach: " << robot_link->GetName() << " found";
-            }
-
-            //This is joint creation
-            gazebo::physics::JointPtr joint;
+            child_model_link = child_model_links[i];
+            yInfo() << LogPrefix << "Attach: " << child_model_link->GetName() << " link found";
+            break;
+        }
+    }
+    if(!child_model_link)
+    {
+    	yError() << LogPrefix << "Attach: " << child_model_link_name << " link is not found";
+    	return false;
+    }
 
-            #if GAZEBO_MAJOR_VERSION >= 8
-            joint = _world->Physics()->CreateJoint("fixed",object_model);
-            #else
-            joint = _world->GetPhysicsEngine()->CreateJoint("fixed",object_model);
-            #endif
+    //This is joint creation
+    gazebo::physics::JointPtr joint;
```

Comment:
I think it i better to use the variable name to `fixedJoint` or `magnetJoint` to be more reflective of its use.